### PR TITLE
feat: add meeting search index

### PR DESCRIPTION
**Pull request by AdrianAndersen**
Add meeting search index for meetings.

This makes it possible to search for meeting titles, meeting descriptions and meeting reports.

Relevant frontend PR: webkom/lego-webapp#2584

### CODE REVIEW

File: lego/apps/meetings/search_indexes.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    serializer_class = MeetingSearchSerializer
+    result_fields = (
+        "title",
+        "description",
+        "report",
+        "start_time",
+    )
+    autocomplete_result_fields = ("title", "start_time")
+
+    autocomplete_fields = ("title", "description", "report")
```
Comment (LudvigHz) [inline lego/apps/meetings/search_indexes.py:21]: Remove `"report"` here. It doesn't make sense to autocomplete based on the content of the meeting report.

**Review by LudvigHz — CHANGES_REQUESTED** [review]
Awesome!

File: lego/apps/meetings/search_indexes.py (hunk 1 above)
Comment (AdrianAndersen) [reply lego/apps/meetings/search_indexes.py:19]: The meeting report is the actual content of the meeting, while the description is the short message in calendars. I think it makes sense to be able to search for something from the report. (aka. the meeting notes). At least in our case, it is not very useful to only search for the meeting title. What do you think?

Comment (LudvigHz) [reply lego/apps/meetings/search_indexes.py:19]: Yes, that's why it makes sense to have it in `search_fields`, which is exactly what this field is for (as you have done). But not in `autocomplete_fields`, as that would just be confusing and not make sense.

Comment (AdrianAndersen) [reply lego/apps/meetings/search_indexes.py:19]: Hm, that is weird. In my testing, I have to include "report" in the autocomplete_fields as well to make it actually show up in the results. 🤔

Comment (LudvigHz) [reply lego/apps/meetings/search_indexes.py:19]: So, the reason for this is kinda weird/stupid. In lego-webapp, when you search in the top bar, and wait for results to show up as you type, we use the `/autocomplete` endpoint, and the full search is only done once you press Enter.

This is kinda strange and confusing, and we have discussed before what the best behavior is here, but haven't changed it, or made it more clear that the search-as-you-type results aren't full search. The behavior I describe we want for meetings, is also how it works for events, pages, articles etc. If you search for page content in lego-webapp now, you'll notice you won't get any results, but when you press enter, you will see the expected results.

Comment (LudvigHz) [reply lego/apps/meetings/search_indexes.py:19]: It does say "Trykk enter for fullstendig søk at the bottom of the results" but it's not clear enough what the behavior is I guess :upside_down_face:.

Comment (AdrianAndersen) [reply lego/apps/meetings/search_indexes.py:19]: Ah yes, I see. Then it makes sense, I agree.

**Review by SmithPeder — APPROVED** [review]

**Review by LudvigHz — APPROVED** [review]

**Pull request merged**